File: Programmers/FunctionDeveloping.cpp
#include <string>
#include <vector>

//프로그래머스 스택/큐 - 기능개발 맞았음!
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> left, timeline;

    for(int i=0; i<progresses.size(); i++){
        int leftwork = 100-progresses[i];
        left.push_back(leftwork);

        int worktime = leftwork/speeds[i];
        if(leftwork%speeds[i] > 0){
            worktime++;
        }
        timeline.push_back(worktime);
    }

    int flag = timeline[0];
    int published = 1;
    for(int i=1; i<timeline.size(); i++){
        if(timeline[i]<=flag){
            published++;
        }else{
            answer.push_back(published);
            flag = timeline[i];
            published = 1;
        }
    }
    answer.push_back(published);
    return answer;
}

int main(){
    vector<int> answer;

    answer = solution({93, 30, 55}, {1, 30, 5});

    return 0;
}

File: Programmers/FunctionDeveloping.swift
import Foundation

// 프로그래머스 기능 개발, level 2
func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    if progresses.count == 1 {
        return [1]
    }
    
    var left_days = Array(repeating: 0, count: progresses.count)
    
    for i in 0..<progresses.count {
        let work = 100 - progresses[i]
        
        var days = work / speeds[i]
        if work%speeds[i] > 0 {
            days += 1
        }
        
        left_days[i] = days
    }
    
    var answer = [Int]()
    var count = 1
    var standard = left_days[0]
    
    for i in 1..<progresses.count {
        if left_days[i] <= standard {
            count += 1
            continue
        }
        
        answer.append(count)
        count = 1
        standard = left_days[i]
    }
    
    answer.append(count)
    return answer
}

print(solution([95, 90, 99, 99, 80, 99],[1, 1, 1, 1, 1, 1]))
